src/sql/azdata.proposed.d.ts

```diff
@@ -1934,4 +1934,21 @@ declare module 'azdata' {
 			isPrimary: boolean;
 		}
 	}
+
+	export interface ModelBuilder {
+		chart(): ComponentBuilder<ChartComponent, ChartComponentProperties>;
+	}
+
+	export interface ChartComponentProperties extends ComponentProperties {
+		chartType?: string | undefined;
+		data?: number[] | undefined;
```

Comment:
I am guess things should be depend on types of charts as well. You can create a separate Interface DoughnutChartData and make the data of that type

for now it should look like data: DoughnutChartData | undefined

Reply:
1. A general comment that also applies to other places: if you use `?` then you don't have do use `| undefined`
2. As @aasimkhan30 pointed out, different chart type might have different properties/options, so I'd recommend to define a base generic interface
```
interface ChartComponentProperties<T> {
    data: T[] | undefined;
    // other common chart properties
}

interface DoughnutChartComponentProperties extends ChartComponentProperties<DoughnutChartData> {
   // Doughnut chart specific options
}

interface DoughnutChartData {

}
```

Reply:
For what it's worth there is a difference, which can matter. `?` is saying "this property is optional so may not exist on the object at all", while `| undefined` is saying "this property must exist but the value may be undefined". 

So you still wouldn't usually combine them, but which one you use depends on what you want the person using the typings to be doing - do you want them to always explicitly add the property but then allow them to have it be undefined? This can help for properties that are important but not required. Or are you fine with them leaving it out (which may mean they never noticed it was a valid property in the first place) in the interest of making it easier to create objects that fit the type.

Typically in our cases having the property be optional (`?`) is what we want so that's a good default, but it's worth taking a few moments to think about how you want the property you're adding to be used.

Reply:
modified it to specific 'doughnutChart' component